packages/graph-retriever/src/graph_retriever/utils/merge.py

```diff
@@ -0,0 +1,71 @@
+import asyncio
+from collections.abc import AsyncIterator
+from typing import TypeVar
+
+from attr import dataclass
+
+T = TypeVar("T")
+
+
+@dataclass
+class _Done:
+    exception: bool = False
+
+
+async def amerge(
```

Comment:
Utility so that I can merge the multiple async iterators into a single async iterator. Couldn't find this in readily available libraries, so threw it in the utilities.